Sources/ProjectDescription/Config.swift

```diff
@@ -75,14 +60,34 @@ public struct Config: Codable, Equatable, Sendable {
         generationOptions: GenerationOptions = .options(),
         installOptions: InstallOptions = .options()
     ) {
-        self.compatibleXcodeVersions = compatibleXcodeVersions
-        self.plugins = plugins
-        self.generationOptions = generationOptions
-        self.installOptions = installOptions
-        self.cloud = cloud
+        var fullHandle = fullHandle
+        var url = url
+        var generationOptions = generationOptions
+        if let cloud {
+            fullHandle = cloud.projectId
+            url = cloud.url
+            generationOptions.optionalAuthentication = cloud.options.contains(.optional)
+        }
+        project = TuistProject.tuist(
+            compatibleXcodeVersions: compatibleXcodeVersions,
+            swiftVersion: swiftVersion,
+            plugins: plugins,
+            generationOptions: generationOptions,
+            installOptions: installOptions
+        )
+        self.fullHandle = fullHandle
+        self.url = url
+        dumpIfNeeded(self)
+    }
+
+    public init(
+        project: TuistProject,
+        fullHandle: String? = nil,
+        url: String = "https://tuist.dev"
```

Comment:
```suggestion
        fullHandle: String? = nil,
        url: String = "https://tuist.dev",
        project: TuistProject
```

nit: I'd probably change the order to keep the `fullHandle` at the top of the init. Alternatively we can do:
```swift
        fullHandle: String? = nil,
        project: TuistProject,
        url: String = "https://tuist.dev"
```
To order the properties alphabetically.